docs/maui/layouts/DockLayout.md

```diff
@@ -0,0 +1,122 @@
+---
+title: DockLayout - .NET MAUI Community Toolkit
+author: profix898
+description: The `DockLayout` is a layout where views can be docked to the sides (top, left, right, bottom) of the layout container.
+ms.date: 11/04/2022
+---
+
+# DockLayout
+
+`DockLayout` is a layout where views can be docked to the sides of the layout container.
+
+The image below shows how a `DockLayout` is conceptually structured. Child views are docked at one of 4 possible docking positions: *Top*, *Bottom*, *Left* or *Right* (equivalent to `DockPosition.Top`, `DockPosition.Bottom`, `DockPosition.Left`, and `DockPosition.Right`). Views that are not explicitly docked (or with `DockPosition.None`) are displayed at the center (or between *Top* / *Bottom* and *Left* / *Right* positions).
+
+![DockLayout concept](/docs/images/layouts/DockLayout.svg)
+
+## Building a DockLayout
+
+A basic `DockLayout` can be created in XAML as shown here:
+
+### XAML
+
+```xml
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:mct="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
+             x:Class="MyProject.MyContentPage">
+
+    <mct:DockLayout>
+        <Button mct:DockLayout.DockPosition="Top" Text="Top" HeightRequest="50" />
+        <Button mct:DockLayout.DockPosition="Bottom" Text="Bottom" HeightRequest="70" />
+        <Button mct:DockLayout.DockPosition="Left" Text="Left" WidthRequest="80" />
+        <Button mct:DockLayout.DockPosition="Right" Text="Right" WidthRequest="90" />
+        <Button Text="Center" />
+    </mct:DockLayout>
+    
+</ContentPage>
+```
+
+For *Left* / *Right* docking, a `WidthRequest` should be specified. For *Top* / *Bottom* docking, a `HeightRequest` defines the size of the child view along the docking direction. The orthogonal directions are always calculated implictly by the `DockLayout` manager.
```

Comment:
Just to confirm is it a should or a must, what happens if no `WidthRequest` is supplied for `DockPosition.Left`?

Reply:
Should (not must): The `LayoutManager` uses the `DesiredSize` of each child view. In case no Height/WidthRequest is set, the view still returns a minimal size it wants to occupy.